autogen/agentchat/conversable_agent.py

```diff
@@ -851,24 +852,34 @@ async def a_initiate_chat(
 
     def _summarize_chat(
         self,
-        method,
-        agent: Optional[Agent] = None,
+        summary_method,
+        recipient: Optional[Agent] = None,
         prompt: Optional[str] = None,
         cache: Optional[Cache] = None,
     ) -> str:
         """Get a chat summary from an agent participating in a chat.
 
         Args:
-            method (str): the method to get the summary.
-            agent: the participating agent in a chat.
+            summary_method (str or callable): the summary_method to get the summary.
+                The callable summary_method should take the recipient and sender agent in a chat as input and return a string of summary. E.g,
```

Comment:
What happens if this is used in the context of a group chat where the messages are in the GroupChat messages structure?

Reply:
The "reflection_with_llm" option is using llm to summarize based on [`chat_messages_for_summary`](https://github.com/microsoft/autogen/blob/main/autogen/agentchat/groupchat.py#L481C9-L481C34) from group chat. 

The added callable takes the sender and recipient as inputs (both could be group chat manager), and the user could decide how to use info from these two agents.